src/graphnet/data/sqlite_dataset.py

```diff
@@ -118,6 +120,30 @@ def _get_dbang_label(self, truth_dict):
         except:
             return -1
 
+    def _get_unique_positions(self, tensor):
+        return torch.unique(tensor, return_counts = True, return_inverse=True, dim=0)
+
+    def _make_dom_wise_representation(self,data):
+        unique_doms, inverse_idx, n_pulses_pr_dom = self._get_unique_positions(data.x[:,[0,1,2,5,6]])
```

Comment:
I would be hesitant to hardcode position, dom number, and string number variables as integer indices. It would probably be more clear to do something like https://github.com/icecube/graphnet/blob/main/src/graphnet/models/detector/icecube.py#L116  (although I admit that hardcoding these as strings is only slightly better than doing so as integers). Also, I don't think that indices `5` and `6` refer to the dom and string number, if that was the intention?
```suggestion
        unique_doms, inverse_idx, n_pulses_pr_dom = self._get_unique_positions(torch.stack((
            data['dom_x'],
            data['dom_y'],
            data['dom_z'],
            data['dom_number'],
            data['string'],
        ), dim=1))
```

Reply:
[0,1,2,5,6] corresponds to ['dom_x', 'dom_y', 'dom_z', 'rde', 'pmt_area' ]. I include 5,6 because they are then included in the `unique_doms` variable, thus limiting the subsequent for loop to only match pulse-related info with the DOM.  The input features are not saved as fields in the graph, so your suggestion will not work. But I will find a solution that removes the hard-coded indicies.